hacspec/examples/aes-gcm/aes.rs

```diff
@@ -90,43 +104,59 @@ fn mix_columns(state: Block) -> Block {
     mix_column(3, state)
 }
 
-fn add_round_key(state: Block, key: Key) -> Block {
+fn add_round_key(state: Block, key: RoundKey) -> Block {
     let mut out = state;
     for i in 0..BLOCKSIZE {
         out[i] ^= key[i];
     }
     out
 }
 
-fn aes_enc(state: Block, round_key: Key) -> Block {
+fn aes_enc(state: Block, round_key: RoundKey) -> Block {
     let state = sub_bytes(state);
     let state = shift_rows(state);
     let state = mix_columns(state);
     add_round_key(state, round_key)
 }
 
-fn aes_enc_last(state: Block, round_key: Key) -> Block {
+fn aes_enc_last(state: Block, round_key: RoundKey) -> Block {
     let state = sub_bytes(state);
     let state = shift_rows(state);
     add_round_key(state, round_key)
 }
 
-fn rounds(state: Block, key: Bytes144) -> Block {
+fn rounds_aes128(state: Block, key: Bytes144) -> Block {
```

Comment:
So this is something that shouldn't be necessary to have two functions for. But we'll need library support for that.